terraform/resource.go

```diff
@@ -500,9 +500,27 @@ func (c *ResourceConfig) get(
 		}
 	}
 
+	if reallyNil(current) {
+		return nil, false
+	}
 	return current, true
 }
 
+func reallyNil(i interface{}) bool {
```

Comment:
`ValueOrZero` in field_reader leads me to believe a method that finds `nil` on `FieldReadResult` could be useful ... `reallyNil` on any `interface{}` feels a bit broad?

Reply:
This is selectively preventing a typed nil dynamic value in an interface from being returned. The tests assert that a returning `nil, true` is invalid, because nil is not technically valid in the legacy configuration. It does however happen due to various circumstances, and while we can't change the behavior, we can mitigate crashes.

Reply:
Helpful, thanks!